msal/src/main/java/com/microsoft/identity/client/claims/RequestedClaimAdditionalInformation.java

```diff
@@ -0,0 +1,60 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.claims;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RequestedClaimAdditionalInformation {
+
+    @SerializedName("essential")
+    private Boolean mEssential = false;
+
+    @SerializedName("values")
+    private List<Object> mValues = new ArrayList<>();
+
+    @SerializedName("value")
+    private Object mValue = null;
```

Comment:
The expectation is that this is going to be _some_ `Object`, serializable to JSON? Should we directly accept a `JSONObject` (not the `Gson` class, but the Android internal class)

Reply:
These should generally be primitives.  Since Java provides a reference type that corresponds to each of it's primitives.  I was hoping to take advantage of that here.